docs/development/developer_guide.md

```diff
@@ -60,22 +65,50 @@ After the cluster is up, the TFJob CRD should be created on the cluster.
 make install
 ```
 
-### Run Operator
-
-Now we are ready to run operator locally:
+### Build Operator Image
+```sh
+make docker-build IMG=my-username/training-operator:my-pr-01
+```
+You can swap `my-username/training-operator:my-pr-01` with whatever you would like.
 
+## Load docker image 
 ```sh
-make run
+ kind load docker-image my-username/training-operator:my-pr-01
+``` 
+
+## Modify operator image with new one
+Update the `newTag` key in `./manifests/overlayes/standalone/kustimization.yaml` with the new image.
+
+Deploy the operator with: 
+```sh 
+kubectl apply -k ./manifests/overlayes/standalone
 ```
+And now we can submit jobs to the operator.
 
-To verify local operator is working, create an example job and you should see jobs created by it.
+You should be able to see a pod for your training operator running in your namespace using
+```commandline
+kubectl get namespace
+kubectl get pods -n YOUR_NAMESPACE_FROM_ABOVE
+kubectl logs -n YOUR_NAMESPACE_FROM_ABOVE YOUR_PODNAME_FROM_ABOVE
+```
+Please make sure to replace `YOUR_NAMESPACE_FROM_ABOVE` with the namespace you are using and `YOUR_PODNAME_FROM_ABOVE` with the pod name you see from the `kubectl get pods` command.
+
+### Test running an operator locally 
+```sh 
+cd ./examples/pytorch/mnist2/
+docker build -f Dockerfile.cpu .
+kubectl create -f ./sample_pytorchjob.yaml
+kubectl describe PyTorchJob
+kubectl get pods -n YOUR_NAMESPACE_FROM_ABOVE
+kubectl events
+```
+### Run Operator
+
+Now we are ready to run operator locally:
 
 ```sh
-cd ./examples/tensorflow/dist-mnist
-docker build -f Dockerfile -t kubeflow/tf-dist-mnist-test:1.0 .
-kubectl create -f ./tf_job_mnist.yaml
+make run
```

Comment:
Did this work fine in your local?

Reply:
Actually yeah. 🤔

Reply:
In my local, the operator is stuck as below. What happens in your local?

```shell
make run -o manifests -o generate -o fmt -o vet
go run ./cmd/training-operator.v1/main.go
2024-04-20T00:59:55+09:00       INFO    setup   Probe endpoints are configured on healthz and readyz
2024-04-20T00:59:55+09:00       INFO    setup   starting manager
2024-04-20T00:59:55+09:00       INFO    starting server {"kind": "health probe", "addr": "[::]:8081"}
2024-04-20T00:59:55+09:00       INFO    setup   Waiting for certificate generation to complete
2024-04-20T00:59:55+09:00       INFO    controller-runtime.metrics      Starting metrics server
2024-04-20T00:59:55+09:00       INFO    controller-runtime.metrics      Serving metrics server  {"bindAddress": ":8080", "secure": false}
2024-04-20T00:59:55+09:00       INFO    cert-rotation   starting cert rotator controller
2024-04-20T00:59:55+09:00       INFO    Starting EventSource    {"controller": "cert-rotator", "source": "kind source: *v1.Secret"}
2024-04-20T00:59:55+09:00       INFO    Starting EventSource    {"controller": "cert-rotator", "source": "kind source: *unstructured.Unstructured"}
2024-04-20T00:59:55+09:00       INFO    Starting Controller     {"controller": "cert-rotator"}
2024-04-20T00:59:55+09:00       INFO    cert-rotation   no cert refresh needed
2024-04-20T00:59:55+09:00       INFO    Starting workers        {"controller": "cert-rotator", "worker count": 1}
2024-04-20T00:59:55+09:00       INFO    cert-rotation   no cert refresh needed
2024-04-20T00:59:55+09:00       INFO    cert-rotation   Ensuring CA cert        {"name": "validator.training-operator.kubeflow.org", "gvk": "admissionregistration.k8s.io/v1, Kind=ValidatingWebhookConfiguration", "name": "validator.training-operator.kubeflow.org", "gvk": "admissionregistration.k8s.io/v1, Kind=ValidatingWebhookConfiguration"}
```

Reply:
I'm getting the same results, I didn't know what to expect so I had assumed this was satisfactory but sounds like it's not. What output should I expect from `make run`?

Reply:
That is expected behavior since we don't set up any certs for webhooks locally.
So, when we want to spin up the operator, we need to create and set up self signed certs manually... but it is hard...

Reply:
So, I would recommend removing this section for now.